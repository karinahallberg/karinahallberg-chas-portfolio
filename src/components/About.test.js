import { render, screen } from '@testing-library/react'
import About from './About'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

describe('About Test h1 and specified button exists', () => {
  test('Check Twitter function is enabled', () => {
    const { getByText } = render(<About />)
    expect(getByText('Twitter'))
  })
})

test('Creating a snapshot', () => {
  const component = renderer.create(
    <a href='https://twitter.com/HallbergKarina'>Twitter</a>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders button link', () => {
  const { getByText, debug } = render(<About />)
  const linkElement = screen.getByText(/GitHub/i)
  expect(linkElement).toBeInTheDocument()
  debug()
})
