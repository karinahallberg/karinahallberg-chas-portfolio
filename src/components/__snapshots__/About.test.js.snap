// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes color when hovered over 1`] = `
<a
  href="https://twitter.com/HallbergKarina"
>
  Twitter
</a>
`;

exports[`Creating a snapshot 1`] = `
<a
  href="https://twitter.com/HallbergKarina"
>
  Twitter
</a>
`;

exports[`Link changes color when hovered 1`] = `
<a
  href="https://twitter.com/HallbergKarina"
>
  Twitter
</a>
`;
